### PR TITLE
Add support of generating symmetric/asymmetric tensor's range for calibration

### PR DESCRIPTION
**Pull request by chilo-ms**
Add support of generating symmetric/asymmetric tensor's range for quantization calibration.

- When setting "symmetric" with either calibration method, `create_calibrator(... ,extra_options={'symmetric': True})`. The final range of tensor will be explicitly set to symmetric to central point "0".

- When setting "asymmetric" with either calibration method, `create_calibrator(... ,extra_options={'symmetric': False})`. the final range of tensor won't be explicitly set to symmetric to central point "0". (For MinMax method, the final range is real minimum value and maximum value. For Entropy/Percentile, the final range is calculated based on the given approach.)

**Motivation and Context**
- If it fixes an open issue, please link to the issue here. https://github.com/microsoft/onnxruntime/issues/10571

### CODE REVIEW

File: onnxruntime/python/tools/quantization/calibrate.py [hunk 1]
```diff
@@ -275,6 +283,7 @@ def __init__(self,
                  op_types_to_calibrate=[],
                  augmented_model_path='augmented_model.onnx',
                  method='percentile',
+                 symmetric=True,
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/calibrate.py:286]: Could you please set it to false by default?

File: onnxruntime/python/tools/quantization/calibrate.py [hunk 2]
```diff
@@ -372,38 +385,41 @@ def __init__(self,
...
+        super(EntropyCalibrater, self).__init__(model, op_types_to_calibrate, augmented_model_path, method=method,
+                                                symmetric=symmetric, num_bins=num_bins, num_quantized_bins=num_quantized_bins)
 
 class PercentileCalibrater(HistogramCalibrater):
     def __init__(self,
                  model,
                  op_types_to_calibrate=[],
                  augmented_model_path='augmented_model.onnx',
                  method='percentile',
+                 symmetric=True,
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/calibrate.py:409]: >True [](http://example.com/codeflow?start=27&length=4)

and also this one

File: onnxruntime/python/tools/quantization/calibrate.py [hunk 3]
```diff
@@ -553,8 +579,16 @@ def compute_percentile(self):
...
             total = hist.sum()
             cdf = np.cumsum(hist/total)
-            idx = np.searchsorted(cdf, percentile/100)
-            thresholds_dict[tensor] = (float(hist_edges[idx]), float(hist_edges[idx]))
+            idx_right = np.searchsorted(cdf, percentile/100)
+            if self.symmetric:
+                thresholds_dict[tensor] = (-float(hist_edges[idx_right]), float(hist_edges[idx_right]))
+            else:
+                idx_left = np.searchsorted(cdf, (1.0 - percentile/100))
+                thresholds_dict[tensor] = (float(hist_edges[idx_left]), float(hist_edges[idx_right]))
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/calibrate.py:587]: should it be left 1.0 - percentile/200 and right: percentile/200?

**Comment by yufenglee** [comment]
Could you please pass the symmetric info to create_calibrator here:
https://github.com/microsoft/onnxruntime/blob/05d6805830ddeff6c4229ade7d1ae07c97d97f9b/onnxruntime/python/tools/quantization/quantize.py#L217

File: onnxruntime/python/tools/quantization/calibrate.py (hunk 3 above)
Comment (chilo-ms) [reply onnxruntime/python/tools/quantization/calibrate.py:588]: Yes, fixed the code.

**Comment by chilo-ms** [comment]
> Could you please pass the symmetric info to create_calibrator here:
> 
> https://github.com/microsoft/onnxruntime/blob/05d6805830ddeff6c4229ade7d1ae07c97d97f9b/onnxruntime/python/tools/quantization/quantize.py#L217

Modified the code to pass symmetric info.

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**